### ISSUE TITLE
node x86 artifact build is broken (caused by c-ares?)

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#10600]
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/3204/architecture=x86,language=node,platform=linux/console

```
collect2: error: ld returned 1 exit status
make: *** [Release/obj.target/deps/cares/cares.so] Error 1
deps/cares/cares.target.mk:179: recipe for target 'Release/obj.target/deps/cares/cares.so' failed
make: Leaving directory '/var/local/git/grpc/node_modules/cares/build'
```

**Comment by jtattermusch** [comment]
CC @y-zeng

**Comment by y-zeng** [comment]
grpc node is using the cares library from https://www.npmjs.com/package/cares. It has a modified version of [ares.h](https://github.com/royalpinto/node-cares/blob/master/deps/cares/include/ares.h), which defines:
```
typedef int ares_socklen_t;
```
This might be the cause of the` -Wpointer-sign` warning. I'm guessing this package is using the pre-defined types to get rid of the generated `ares_build.h`. @murgatroid99 Should we silence this warning?

**Comment by murgatroid99** [comment]
It looks to me like the failure actually comes from attempting to install that npm package on 32 bit Linux. That's before we even start any gRPC compilation.

**Comment by murgatroid99** [comment]
The actual error there is this part

```
  SOLINK(target) Release/obj.target/deps/cares/cares.so
/usr/bin/ld: skipping incompatible /usr/lib/gcc/i586-linux-gnu/4.9/libstdc++.so when searching for -lstdc++
/usr/bin/ld: skipping incompatible /usr/lib/gcc/i586-linux-gnu/4.9/libstdc++.a when searching for -lstdc++
/usr/bin/ld: cannot find -lstdc++
collect2: error: ld returned 1 exit status
```

**Comment by murgatroid99** [comment]
This seems to be fixed by #10744.
